app/controllers/application_controller/advanced_search.rb

```diff
@@ -48,7 +48,7 @@ def adv_search_button_saveid
       add_flash(_("Search Name is required"), :error) if params[:button] == 'saveit'
       false
     else
-      s = @edit[@expkey].build_search(@edit[:new_search_name], @edit[:search_type], session[:userid])
+      s = @edit[@expkey].build_search(@edit[:new_search_name], current_user.super_admin_user? || current_user.super_admin_user? ? @edit[:search_type] : nil, session[:userid])
```

Comment:
Why is `current_user.super_admin_user? || current_user.super_admin_user? ? @edit[:search_type] : nil` needed here?

Reply:
It solves the following bug:

User can load a custom filter made by  the admin ..
![image](https://user-images.githubusercontent.com/24387870/66042199-1f63f400-e51c-11e9-9703-e28774bf93b8.png)

.. and save a changed version of this.
![image](https://user-images.githubusercontent.com/24387870/66042213-2a1e8900-e51c-11e9-85cf-2a587041bc8f.png)

Now the filter will be seen by all users, not only of his own tenant.
![image](https://user-images.githubusercontent.com/24387870/66042227-31459700-e51c-11e9-9442-d5bdfaf4a251.png)

This is because the loaded filter will adopt the search type of the used filter. So when a user loads an global filter, his changed version will also be global. This proposed change fixes this, by only allowing admins and superadmins to save filters as global filters

Reply:
Right. But why would you need to do
```ruby
current_user.super_admin_user? || current_user.super_admin_user?
```
?

i.e. why do you need `true || true` (or `false || false`) here?

Reply:
Oops.. Of course that had to be admin and super_admin. I overlooked it both after your comment and after writing it.